internal/controller/http/restrouter.go

```diff
@@ -110,32 +104,36 @@ func (c *RestController) sendResponse(
 		data, err := json.Marshal(response)
 		if err != nil {
 			c.lc.Error(fmt.Sprintf("Unable to marshal %s response", api), "error", err.Error(), common.CorrelationHeader, correlationID)
-			http.Error(writer, err.Error(), http.StatusInternalServerError)
-			return
+			// set Response.Committed to true in order to rewrite the status code
+			writer.Committed = false
+			return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
 		}
 
 		_, err = writer.Write(data)
 		if err != nil {
 			c.lc.Error(fmt.Sprintf("Unable to write %s response", api), "error", err.Error(), common.CorrelationHeader, correlationID)
-			http.Error(writer, err.Error(), http.StatusInternalServerError)
-			return
+			// set Response.Committed to true in order to rewrite the status code
```

Comment:
Comment says `true` but code sets to `false`. Which is correct?
Same for similar comment above and below

Reply:
Updated the comments to sync with code `set Response.Committed to false`.